servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -244,37 +238,41 @@ def _get_json_batch(self, url, offset, batch_size, timeout, auth):
         limited_url = url + limit_args
         return self._get_json(limited_url, timeout, auth)
 
-    def _get_json(self, url, timeout, auth=None, verify=True):
-        tags = ["url:%s" % url]
-        msg = None
-        status = None
-        resp = None
-        try:
-            resp = requests.get(url, timeout=timeout, auth=auth, verify=verify)
-            if resp.status_code != 200:
-                status = AgentCheck.CRITICAL
-                msg = "Got %s when hitting %s" % (resp.status_code, url)
-        except requests.exceptions.Timeout:
-            # If there's a timeout
-            msg = "%s seconds timeout when hitting %s" % (timeout, url)
-            status = AgentCheck.CRITICAL
-        except Exception as e:
-            msg = str(e)
-            status = AgentCheck.CRITICAL
-        finally:
-            if status is AgentCheck.CRITICAL:
-                self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags,
-                                   message=msg)
-                raise CheckException("Cannot connect to ServiceNow CMDB, please check your configuration.")
-
-        if resp.encoding is None:
-            resp.encoding = 'UTF8'
+    @staticmethod
+    def _get_json(url, timeout, auth=None, verify=True):
+        execution_time_exceeded_error_message = 'Transaction cancelled: maximum execution time exceeded'
+
+        response = requests.get(url, timeout=timeout, auth=auth, verify=verify)
+        if response.status_code != 200:
+            raise CheckException("Got %s when hitting %s" % (response.status_code, url))
+
         try:
-            resp = yaml.safe_load(resp.text.encode("utf-8"))
-        except Exception as e:
-            self.log.exception(str(e))
-            raise CheckException("Exception occurred while parsing response and the error is : {}".format(str(e)))
+            response_json = yaml.safe_load(response.text.encode('utf-8'))
+        except ParserError as e:
+            # Fix for ServiceNow bug: Sometimes there is a response with status 200 and malformed json with
+            # error message 'Transaction cancelled: maximum execution time exceeded'.
+            # We send right error message because ParserError is just side effect error.
+            if execution_time_exceeded_error_message in response.text:
+                raise CheckException(execution_time_exceeded_error_message)
+            else:
+                raise e
 
-        if resp.get("error"):
-            raise CheckException("Problem in collecting CIs : {}".format(resp["error"].get("message")))
-        return resp
+        if response_json.get("error"):
+            raise CheckException(response_json["error"].get("message"))
+
+        return response_json
+
+    @staticmethod
+    def _get_mandatory_instance_values(instance):
```

Comment:
Instead of this why don't we use something like `schematics` to validate the models that we expect? Then we can validate the value and type for instance

Reply:
I was wondering if `schematics` project is still alive as last release on PyPI was in June 2018 and last commit to master December 2018. Then I found this https://github.com/schematics/schematics/issues/616
It seems that someone is still maintaining it.
OK, I'll introduce `schematics` to agent check.

Reply:
@zandrewitte I used schematics to validate instance data. Please take a look if that is OK.
I would like to use it on a few more places like checking if result list exist in json response.